### PR TITLE
CI: Add i18n verification step

### LINKED ISSUE #61301: Additional checks on i18n PRs
We should add additional checks to PRs related to internationalization in the Grafana OSS codebase to ensure that strings have been correctly and completely marked up for translation. With the increasing number of community contributions to internationalization, this will help to ensure consistency across the codebase.

The check should make sure developers have not marked up phrases incorrectly (dynamic phrases, or duplicate keys), they have ran `yarn i18n:extract` and comitted the result:

 - Do the check in the Drone pipeline
 - The check should run `yarn i18n:extract` and check that it runs successfully (returns 0 status code)
   - Verify that the extract returns non-zero status code when it encounters these errors:
     - A translation key is used twice
     - A translation phrase contains a dynamic expression (such as `t("preferences.theme.${themeID}", themeName)` that cannot be statically extracted
  - The check should make sure the `public/locales/*/grafana.json` translation files have no uncomitted changes after running `yarn i18n:extract`
  - The check should include helpful error messages so contributors know how to fix the problem themselves, and they aren't discouraged or have to ask for additional help.

[See also the i18n contribution guidelines.](https://github.com/grafana/grafana/blob/813ebf988d354bf23e981e00c9c8458995774c21/contribute/internationalization.md)

### PR DESCRIPTION
**Pull request by tskarhed**
**What is this feature?**

Adds a step for verifying i18n, with compiling and extracting

**Why do we need this feature?**

To prevent having translation inconsistencies (code vs. what's in the grafana.json files) in main. An example for such a fix can be found in #69381

**Who is this feature for?**
Making it more reliable to review external translation PRs

**Which issue(s) does this PR fix?**:

Fixes #61301

**Special notes for your reviewer:**

1. Build that succeeds https://drone.grafana.net/grafana/grafana/119512/4/6
2. Build that fails because of dynamic key https://drone.grafana.net/grafana/grafana/119481/4/6
3. Build that fails on multiple keys https://drone.grafana.net/grafana/grafana/119514/4/6
4. Build that fails when extracted files have not been commited https://drone.grafana.net/grafana/grafana/119743/4/6

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: scripts/drone/steps/lib.star [hunk 1]
```diff
@@ -775,11 +775,22 @@ def lint_frontend_step():
...
             "yarn run lint",
-            "yarn run i18n:compile",  # TODO: right place for this?
             "yarn run typecheck",
         ],
     }
 
+def verify_i18n_step():
+    return {
+        "name": "verify-i18n",
+        "image": build_image,
```
Comment (tskarhed) [inline scripts/drone/steps/lib.star:785]: I noticed that the other frontend steps have 
```
        "environment": {
            "TEST_MAX_WORKERS": "50%",
        },
```
Is this something needed here too?

File: scripts/drone/steps/lib.star [hunk 2]
```diff
@@ -775,11 +775,32 @@ def lint_frontend_step():
...
+        "commands": [
+            "yarn run i18n:extract",
+            # Verify that translation extraction has been committed
+            '''
+            file_diff=$(git diff -- locales --stat)
+            if [ -n "$file_diff" ]; then
+                echo $file_diff
+                echo "Translation extraction has not been committed. Please commit the changes and push again."
+                exit 1
+            fi
```
Comment (tskarhed) [inline scripts/drone/steps/lib.star:798]: This whole part is visible in the the output regardless if it succeeds or fails. Should it be extracted to a script file?

File: scripts/drone/steps/lib.star [hunk 3]
```diff
@@ -775,11 +775,34 @@ def lint_frontend_step():
         "commands": [
             "yarn run prettier:check",
             "yarn run lint",
-            "yarn run i18n:compile",  # TODO: right place for this?
             "yarn run typecheck",
         ],
     }
 
+def verify_i18n_step():
+    extract_error_message = "Extraction failed. Make sure that you have no dynamic translation phrases, such as 't(\\`preferences.theme.\\$\\${themeID}\\`, themeName)' and that no translation key is used twice."
```
Comment (tskarhed) [inline scripts/drone/steps/lib.star:783]: This is making me confused.
1. Escape character for Starlark
2. Escape character for Drone
3. Escape character for Bash

File: scripts/drone/steps/lib.star [hunk 4]
```diff
@@ -775,11 +775,34 @@ def lint_frontend_step():
         "commands": [
             "yarn run prettier:check",
             "yarn run lint",
-            "yarn run i18n:compile",  # TODO: right place for this?
             "yarn run typecheck",
         ],
     }
 
+def verify_i18n_step():
+    extract_error_message = "Extraction failed. Make sure that you have no dynamic translation phrases, such as 't(\\`preferences.theme.\\$${themeID}\\`, themeName)' and that no translation key is used twice."
```
Comment (tskarhed) [inline scripts/drone/steps/lib.star:783]: Just some interesting observation with `\\$${themeID}`
After Starlark:
```
\$${themeID}
```
After Drone:
```
\${themeID}
```
After bash:
```
${themeID}
```

Comment (joshhunt) [reply scripts/drone/steps/lib.star:783]: argh!

Comment (joshhunt) [inline scripts/drone/steps/lib.star:783]: What do you think about adding a newline at the beginning of this message (I presume `\n` at the beginning will do the job) to help make the help message stand out a bit more?

Also annoying that the output makes it difficult to see which file has the error - maybe we could add an instruction to the file like `Search the output for `[warning] to find the offending file.`

File: public/app/core/components/Footer/Footer.tsx [hunk 5]
```diff
@@ -18,7 +18,7 @@ export let getFooterLinks = (): FooterLink[] => {
     {
       target: '_blank',
       id: 'documentation',
-      text: t('nav.help/documentation', 'Documentation'),
+      text: t(`nav.help/documentation`, 'Documentation'),
```
Comment (joshhunt) [inline public/app/core/components/Footer/Footer.tsx:21]: remember to undo this change just to keep the blame a bit easier to grok 😌

**Review by joshhunt — COMMENTED** [review]
Excellent stuff! I really appreciate the testing and links to the failing builds so we can review them properly 😌 

I'll leave @grafana/grafana-delivery for the actual ✅

File: scripts/drone/steps/lib.star (hunk 4 above)
Comment (tskarhed) [reply scripts/drone/steps/lib.star:783]: I agree, I do however think that some of this error handling/messages shouldn't be in the CI, but rather directly in the commands, so the help can be displayed as early as possible. But we can go with the fix you mentioned for now.

**Review by zerok — APPROVED** [review]
I think it's a bit weird that the i18n:compile step has always been somewhere around the linting stage, but that's unrelated to this change 👍
